### ISSUE TITLE
ambiguous error in node16

### ISSUE THREAD

**Issue by amundsen-roald** [sequelize/sequelize#13999]
### What are you doing?

I'm trying to join one-to-many like this:

- model

```js
model.Module.belongsTo(model.ModuleDetail, {
    foreignKey: 'md_id',
    targetKey: 'md_id',
    as: 'detail',
  })
```

- query
```js
model.Module.findAll({
  include: [
    {
      model: model.ModuleDetail,
      as: 'detail',
    },
  ],
})
```

### What do you expect to happen?

maybe the SQL would be given like:

```sql
SELECT
`bar`.`md_id`,
`bar`.`md_name`,
`detail`.`id` AS `detail.id`,
`detail`.`md_id` AS `detail.md_id`,
`detail`.`img` AS `detail.img`
FROM `module` AS `bar`
LEFT OUTER JOIN `module_detail` AS `detail`
ON `bar`.`md_id` = `detail`.`md_id`;
```

### What is actually happening?

actually I get:

```sql
SELECT
``.`md_id`,
``.`md_name`,
`detail`.`id` AS `detail.id`,
`detail`.`md_id` AS `detail.md_id`,
`detail`.`img` AS `detail.img`
FROM `module` AS ``
LEFT OUTER JOIN `module_detail` AS `detail`
ON ``.`md_id` = `detail`.`md_id`;
```

and there is an error like:```Column 'md_id' in field list is ambiguous.```

### Additional context

I think the empty alias is the reason why I get this error.

But when I switch the node version to 10, it won't happend.

### Environment

- Sequelize version: 6.14.0
- Node.js version: 16.13.2
- Operating System: windows10,  mac os catalina
- TypeScript version: 4.5.5

### How does this problem relate to dialects?

- [x] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [x] No, I don't have the time and I wouldn't even know how to start.

**Comment by WikiRik** [comment]
Which SQL dialect are you using?

**Comment by amundsen-roald** [comment]
> 

standard SQL

**Comment by amundsen-roald** [comment]
> Which SQL dialect are you using?

standard SQL

**Comment by WikiRik** [comment]
Sorry, let me rephrase that question. Which kind of database dialect are you using? Postgres, MySQL, SQLite, etc.

**Comment by amundsen-roald** [comment]
> Sorry, let me rephrase that question. Which kind of database dialect are you using? Postgres, MySQL, SQLite, etc.

It's mysql 😅

**Comment by amundsen-roald** [comment]
> Sorry, let me rephrase that question. Which kind of database dialect are you using? Postgres, MySQL, SQLite, etc.

I used the eggjs framework and found out that the original egg-sequelize problem caused, thanks !

**Comment by WikiRik** [comment]
Ah, so this is not something we can fix?

**Comment by mecoepcoo** [comment]
It maybe problem with `egg-sequelize`, but i have no idea.

I have the same problem with egg & `egg-sequelize` (used sequelize: ^6.0.0). I run the same code in nestjs & `@nestjs/sequelize` and the problem is gone.

**Comment by amundsen-roald** [comment]
> egg-sequelize

Yes, I had problems when I used the ```egg-sequelize``` dependency, but no problems with ```@nestjs/sequelize```, it just works !

**Comment by WikiRik** [comment]
Okay, I will close this issue then since it's not on our end. Feel free to submit a bug report if you find the issue for egg-sequelize. Do note that because of #14003 this issue might be fixed in sequelize v6.14.1

**Comment by amundsen-roald** [comment]
> Okay, I will close this issue then since it's not on our end. Feel free to submit a bug report if you find the issue for egg-sequelize. Do note that because of #14003 this issue might be fixed in sequelize v6.14.1

Okay, I will pay attention to that issue, thanks. 😃
